### DISCUSSION TITLE
Welcome to equal-access Discussions!

### DISCUSSION THREAD

**Discussion by tombrunet** [IBMa/equal-access#773] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by amberistheone** [comment]
Hi Tom - 
I was at CSUN last week and attended a number of IBM presentations, all of them wonderful.  I'm really interested in when the visual tab stop tool will be added to the Accessibility Checker.

Amber Gehrts

**Reply by tombrunet** [reply]
As a general rule, we don't give dates for planned features. But, it is basically at the top of our todo list. I think our designers wanted to make a couple of improvements first.

**Reply by amberistheone** [reply]
Makes sense, thank you for the response!

**Comment by jchan-suse** [comment]
Hi Tom,

I am using equal-access with cypress. Is it a way to convert the JSON output from the accessibility test to a more readable format like the chrome extension output  which is in HTML report format? 

Best,
Jason

**Reply by tombrunet** [reply]
Hi - Sorry that I didn't see this. You can set an output format of HTML in the checker configuration. For Cypress, you have to create a cypress.json like:
```
{
  "env": {
    "outputFormat": ["json", "html"]
  }
}
```

Other config options at https://github.com/IBMa/equal-access/blob/master/accessibility-checker/src/README.md#configuration

**Comment by ashishverma012** [comment]
Hi @tombrunet ,

Need help here, I am using npm accessibility-checker package (https://www.npmjs.com/package/accessibility-checker) using Javacsript and I am able to extract violations such as ruleId, message etc but I have requirement to get wcag level (such as A, AA etc) and Checkpoint (such as 2.1.1 Keyboard, 1.1.1 Non-text Content etc). I tried many things but not able to get these parameters. Please assist for this issue, Using below piece of code to get violations -

const puppeteer = require('puppeteer');
const aChecker = require('accessibility-checker');

async function runAccessibilityChecker(url, lastSegment, summaryData, reportDir) {
const browser = await puppeteer.launch();
const page = await browser.newPage();
await page.setCacheEnabled(false);
await page.goto(url, { waitUntil: 'networkidle0', timeout: 120000 });

// Get page content
const pageHTML = await page.content();
await browser.close();

// Run accessibility-checker
const results = await aChecker.getCompliance(pageHTML, 'TestPage');

// Flatten relevant fields for CSV
const violations = results.report.results.map(item => ({
ruleId: item.ruleId,
message: item.message,
level: item.level,
reasonId: item.reasonId,
path: item.path.dom,
snippet: item.snippet,
Element: getElement(item.snippet),
help: item.help
}));
}

const getElement = snippet => {
if (!snippet) return '';
const match = snippet.match(/^<([a-zA-Z0-9-]+)/);
return match ? match[1] : '';
};
